pcsx2/ps2/PGPU.cpp

```diff
@@ -0,0 +1,1110 @@
+			triggerPgifInt(0);
+			ringBufPut(&pgifCmdRbC, &data);
+		}
+		#if CMD_INTERCEPT_AT_WR == 1
+		ckhCmdSetPgif(data);
+		#endif
+	}
+	return;
+}
+
+u32 psxGPUr(int addr) {
+	u32 data;
+	if (addr == HW_PS1_GPU_DATA) {
+		//data = PGpuDataReg;
+		//The following is commented-out, because it seems more logical to return GP1() CMD responces only when Data FIFO is empty.
+		//if ((lastGpuCmd & 0xFF000000) == 0x10000000) {
+		//	data = immRespHndl(lastGpuCmd, PGpuDataReg);
+		//	lastGpuCmd = 0;
+		//} else {
+			datRingBufGet(&data);
+		//}
+	} else if (addr == HW_PS1_GPU_STATUS) {
+		data = getUpdPgpuStatReg();
+	}
+	#if REG_LOG == 1
+	if (addr != HW_PS1_GPU_STATUS) //prints way too many times (pools)
+		Console.WriteLn( "PGPU read  0x%08X = 0x%08X  EEpc= %08X ",  addr, data, cpuRegs.pc);
+	#endif
+	return data;
+}
+
+//PGIF registers I/O handlers:
+
+void PGIFw(int addr, u32 data) {
+	#if REG_LOG == 1
+	if (addr != PGIF_CTRL)
+	if ((addr != PGIF_CTRL) || ((addr == PGIF_CTRL) && (getUpdPgifCtrlReg() != data)))
+	if (addr != PGPU_STAT)
+		Console.WriteLn( "PGIF write 0x%08X = 0x%08X  0x%08X  EEpc= %08X  IOPpc= %08X ",  addr, data, getUpdPgifCtrlReg(),  cpuRegs.pc, psxRegs.pc);
+	#endif
+
+	if (addr == PGPU_CMD_FIFO) {
+		Console.WriteLn( "PGIF CMD FIFO write by EE (SHOULDN'T HAPPEN) 0x%08X = 0x%08X",  addr, data);
+	} else if (addr == PGPU_DAT_FIFO) { //reverse write - mind the direction set by reg f380
+		//PGpuDataReg = data;
+		ringBufPut(&pgifDatRbC, &data);
+	//	Console.WriteLn( "\n\r PGIF REVERSE !!! DATA write 0x%08X = 0x%08X  IF_CTRL= %08X   PGPU_STAT= %08X  CmdCnt 0x%X \n\r",  addr, data,  getUpdPgifCtrlReg(),  getUpdPgpuStatReg(), pgifCmdRbC.count);
+		drainPgpuDmaNrToIop();
+	} else if (addr == PGPU_STAT) {
+		PGpuStatReg = data; //Should all bits be writable?
+	} else if (addr == PGIF_CTRL) { //PGIF config
+		setPgifCtrlReg(data);
+		PGifCtrlRegOld = PGifCtrlReg; //data
+		fillFifoOnDrain(); //Now this checks the 0x8 bit of the PGIF_CTRL reg, so  it here too,
+		//so that it gets updated immediately once it is set.
+	} else if (addr == PGIF_1) {
+		pgif1reg = data; //GP0(E2h)
+	} else if (addr == PGIF_2) {
+		pgif2reg = data;
+	} else if (addr == PGIF_3) {
+		pgif3reg = data;
+	} else if (addr == PGIF_4) {
+		pgif4reg = data;
+	}
+	return;
+}
+
+u32 PGIFr(int addr) {
+	u32 data;
+	if (addr == PGPU_CMD_FIFO) {
+		cmdRingBufGet(&data);
+	} else 	if (addr == PGPU_DAT_FIFO) {
+		fillFifoOnDrain();
+		datRingBufGet(&data);
+	} else 	if (addr == PGPU_STAT) {
+		data = PGpuStatReg;
+	} else 	if (addr == PGIF_CTRL) { //PGIF config
+		data = getUpdPgifCtrlReg();
+	} else if (addr == PGIF_1) {
+		data = pgif1reg; //GP0(E2h)
+	} else if (addr == PGIF_2) {
+		data = pgif2reg;
+	} else if (addr == PGIF_3) {
+		data = pgif3reg;
+	} else if (addr == PGIF_4) {
+		data = pgif4reg;
+	}
+
+	#if REG_LOG == 1
+	if (addr != PGPU_DAT_FIFO)
+	if ((addr != PGIF_CTRL) || ((addr == PGIF_CTRL) && (getUpdPgifCtrlReg() != data)))
```

Comment:
Indeed. Personally, I don't need that log but it feels like the 2nd addr == was meant to be addr != *something*.
In any case, this could just go.